api/src/services/survey-service.ts

```diff
@@ -816,4 +816,23 @@ export class SurveyService extends DBService {
   async deleteOccurrenceSubmission(submissionId: number): Promise<number> {
     return this.surveyRepository.deleteOccurrenceSubmission(submissionId);
   }
+
+  async getSurveyHasUnpublishedContent(surveyId: number): Promise<boolean> {
+    const has_unpublished_attachments = await this.historyPublishService.hasUnpublishedSurveyAttachments(surveyId);
+
+    const has_unpublished_reports = await this.historyPublishService.hasUnpublishedSurveyReports(surveyId);
+
+    const has_unpublished_observations = await this.historyPublishService.hasUnpublishedObservation(surveyId);
+
+    const has_unpublished_summary_results = await this.historyPublishService.hasUnpublishedSummaryResults(surveyId);
+
+    // Is true when survey have unpulished attachments or reports or observations or summary results
```

Comment:
Nit: Move this comment into JS doc for this method

Reply:
done